subsys/usb/class/cdc_acm.c

```diff
@@ -638,7 +638,7 @@ static int cdc_acm_irq_rx_ready(const struct device *dev)
 {
 	struct cdc_acm_dev_data_t * const dev_data = DEV_DATA(dev);
 
-	if (dev_data->rx_ready) {
+	if (dev_data->rx_irq_ena && dev_data->rx_ready) {
```

Comment:
This line should not be changed, it is legitimate to detect if the data is pending even if rx-irq is disabled. I wonder how you get the spurious interrupts when this API functions(https://github.com/zephyrproject-rtos/zephyr/blob/main/include/drivers/uart.h#L877 , https://github.com/zephyrproject-rtos/zephyr/blob/main/include/drivers/uart.h#L982) must be called from interrupt handler. For TX it probably makes sense, similar as it is with some UART drivers.

Reply:
Ah okay, I agree that we can just change this for the TX case then. That makes sense, especially after taking a look at the other drivers. Thanks for the clarification. I think I was only hitting a spurious call in the transmitting case.